### PR TITLE
Recursive loader refactor

### PR DESCRIPTION
**Pull request by dfreedm**
Overhaul of the style loader to handle recursive `@import` rules.

Also added tests.

### CODE REVIEW

File: src/loader.js [hunk 1]
```diff
@@ -56,31 +54,82 @@ function createStyleElement(cssText, scope) {
...
-// generalized loader that's built out of the one in the HTMLImports polyfill.
-// polyfill the loading of a style element's @import via xhr
 function xhrLoadStyle(style, callback) {
-  HTMLImports.xhr.load(atImportUrlFromStyle(style), function (err, resource,
-      url) {
-    replaceAtImportWithCssText(this, url, resource);
-    this.textContent = urlResolver.resolveCssText(this.textContent, url);
-    callback && callback(this);
-  }, style);
+  var tc = style.textContent;
```
Comment (sorvell) [inline src/loader.js]: the variables here are too abbreviated for readability: tc should be textContent, etc.

File: src/loader.js [hunk 2]
```diff
@@ -56,31 +54,82 @@ function createStyleElement(cssText, scope) {
...
+  if (scope.flags.debug || scope.flags.bust) {
+    url += '?' + Math.random();
+  }
+  request.open('GET', url, true);
+  request.send();
+  return request;
 }
 
-var atImportRe = /@import\s[(]?['"]?([^\s'";)]*)/;
+function recursiveFlatten(tc, mc, idx, callback) {
```
Comment (sorvell) [inline src/loader.js]: Needs a comment explaining what this is doing.

Comment (sorvell) [inline src/loader.js]: Is the idx needed explcitly? it's part of the mc thing isn't it? (there's an abbreviation issue again)

File: test/html/styling/style-import.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  <meta charset="UTF-8">
+  <title>Styling via @import</title>
+  <script src="../../../platform.js"></script>
+  <script src="../../../../tools/test/htmltest.js"></script>
+  <script src="../../../../tools/test/chai/chai.js"></script>
+</head>
+<body>
+  <div id="target"></div>
+  <style id="flatten">
+    @import "rules/top.css";
```
Comment (sorvell) [inline test/html/styling/style-import.html]: needs to test multiple @imports in a style element.

**Comment by dfreedm** [comment]
@sorvell PTAL

**Comment by sorvell** [comment]
LGTM

**Pull request merged**